services/brig/src/Brig/Provider/DB.hs

```diff
@@ -407,15 +407,25 @@ paginateServiceTags :: MonadClient m
     => QueryAnyTags 1 3
     -> Maybe Name
     -> Int32
+    -> Maybe ProviderId
     -> m ServiceProfilePage
-paginateServiceTags tags start size = liftClient $ do
+paginateServiceTags tags start size providerFilter = liftClient $ do
     let start' = Name (maybe "" (toLower . fromName) start)
     let size'  = size + 1
     let tags'  = unpackTags tags
-    p <- filterPrefix (fromName start') <$> queryAll start' size' tags'
+    p <- maybeFilterbyProvider providerFilter . filterPrefix (fromName start') <$> queryAll start' size' tags'
     r <- mapConcurrently resolveRow (result p)
     return $! ServiceProfilePage (hasMore p) (catMaybes r)
   where
+    maybeFilterbyProvider :: Maybe ProviderId -> Page TagRow -> Page TagRow
```

Comment:
you could also define `filterByProvider :: ProviderId -> Page TagRow -> Page TagRow`, then `maybeFilterbyProvider = maybe id filterByProvider`. Not sure if that reads better?

Reply:
Thought the same... I made a little change though, think it reads better now